tracer.go

```diff
@@ -22,26 +23,4 @@ type Tracer interface {
 	//         SetTag("lucky_number", 42)
 	//
 	StartTrace(operationName string) Span
-
-	// Like `StartTrace`, but the returned `Span` is made a child of `parent`.
-	//
-	// The `parent` parameter can either be a `context.Context` or an
-	// `opentracing.TraceContext`. In the former case, the implementation
-	// attempts to extract an `opentracing.Span` using `SpanFromGoContext()`.
-	JoinTrace(operationName string, parent interface{}) Span
```

Comment:
I agree that the JoinTrace function is maybe not needed any more. But I think all creation functions for a Span should be collected here in the Tracer. To have the StartChild function in the Span mixes the creation and behavior responsibility of the Span. I vote to move the StartChild into here, e.g. StartTrace(operationName, Span).